packages/ruleset/test/property-inconsistent-name-and-type.test.js

```diff
@@ -0,0 +1,394 @@
+                    value: {
+                      type: 'integer',
+                      description: 'Integer value'
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      };
+
+      testDocument.paths['/v1/songs'] = {
+        post: {
+          requestBody: {
+            content: {
+              'application/json': {
+                schema: {
+                  type: 'object',
+                  description: 'song schema',
+                  properties: {
+                    value: {
+                      type: 'boolean',
+                      description: 'Boolean value'
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      };
+
+      const results = await testRule(
+        name,
+        propertyInconsistentNameAndType,
+        testDocument
+      );
+
+      expect(results).toHaveLength(0);
+    });
+  });
+
+  describe('Should yield errors', () => {
+    it('should error if two properties have the same name but different types', async () => {
+      const testDocument = makeCopy(rootDocument);
+      testDocument.paths['/v1/books'] = {
+        post: {
+          requestBody: {
+            content: {
+              'application/json': {
+                schema: {
+                  type: 'object',
+                  description: 'book schema',
+                  properties: {
+                    running_time: {
+                      type: 'string',
+                      description: 'Running time of the audiobook form.'
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      };
+
+      const results = await testRule(
+        name,
+        propertyInconsistentNameAndType,
+        testDocument
+      );
+
+      expect(results).toHaveLength(2);
+
+      let validation = results[0];
+      expect(validation.code).toBe(name);
+      expect(validation.message).toBe(
+        'Properties with the same name have inconsistent types: running_time.'
+      );
+      expect(validation.path.join('.')).toBe(
+        'paths./v1/movies.post.requestBody.content.application/json.schema.properties.running_time'
+      );
+      expect(validation.severity).toBe(expectedSeverity);
+
+      validation = results[1];
+      expect(validation.code).toBe(name);
+      expect(validation.message).toBe(
+        'Properties with the same name have inconsistent types: running_time.'
+      );
+      expect(validation.path.join('.')).toBe(
+        'paths./v1/books.post.requestBody.content.application/json.schema.properties.running_time'
+      );
+      expect(validation.severity).toBe(expectedSeverity);
+    });
+
+    // due to the nature of the rule initializing the "visitedProperties" object one time, this rule MUST
+    // use a different property to flag than the previous test, or the first instance won't be reported
```

Comment:
Could we avoid this problem if we explicitly set the rule's "visitedProperties" field to {} at the beginning of this test?

Reply:
I don't actually think we can do that but this made me realize we might be able to reset the function options within the test, I'll try that.

Reply:
I learned that we can reset the loaded modules using `jest` and so I've now configured it to do that after each test. This is still slightly unusual-looking but it isolates the workaround to a small snippet of code in the test file only. I think this is the best solution.